operator/pkg/tpath/tpath_test.go

```diff
@@ -153,6 +153,24 @@ a:
     - v2
     - v3_regex
   test: foo
+`,
+		},
+		{
+			desc:      "AddListEntry",
+			path:      `a.b.[name:n2].list`,
+			value:     `foo`,
+			wantFound: true,
+			want: `
+a:
+  b:
+  - name: n1
+    value: v1
+  - name: n2
```

Comment:
maybe there is a limitation here, we can only locate the first element that matches the name.